### PR TITLE
Configure jvmToolchain for Android and Kotlin projects

### PR DESCRIPTION
**Pull request by SimonMarquis**
There was only a single Kotlin JVM project, so I did not create a dedicated convention plugin for that.
But this can be done if needed.

References #571

### CODE REVIEW

File: core/model/build.gradle.kts [hunk 1]
```diff
@@ -19,6 +19,10 @@ plugins {
     id("kotlin")
 }
 
+kotlin {
+    jvmToolchain(11)
```
Comment (lihenggui) [inline core/model/build.gradle.kts:23]: Why was this line of code added in the `core/model` module?

Comment (SimonMarquis) [reply core/model/build.gradle.kts:23]: Because by default this kotlin module uses version 1.8 as you saw in the logs.

**Comment by mmoczkowski** [comment]
Hey @borisf, would you mind checking this out?

File: build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/KotlinAndroid.kt [hunk 2]
```diff
@@ -74,3 +76,12 @@ internal fun Project.configureKotlinAndroid(
 fun CommonExtension<*, *, *, *>.kotlinOptions(block: KotlinJvmOptions.() -> Unit) {
     (this as ExtensionAware).extensions.configure("kotlinOptions", block)
 }
+
+/**
+ * Configure Kotlin's jvm toolchain for Android projects
+ */
+internal fun Project.configureKotlinAndroidToolchain() {
+    extensions.configure<KotlinAndroidProjectExtension> {
+        jvmToolchain(11)
```
Comment (borisf) [inline build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/KotlinAndroid.kt:85]: This is our recommendation to use Java bytecode version 11

**Review by borisf — COMMENTED** [review]
Looks good to me

**Review by mmoczkowski — CHANGES_REQUESTED** [review]
Simon, could you please address the conflicts? Afterwards we're good to go. Thanks a lot! 😄

**Review by mmoczkowski — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**